### DISCUSSION TITLE
Attack behavior for controllers

### DISCUSSION THREAD

**Discussion by fwannmacher** [diasurgical/DevilutionX#4016] [Ideas]
Currently the implementation for attacking with a controller is just emulating a mouse click when the button is pressed and we have to keep pressing the button for each attack. My suggestion is that we could change it for keep attacking as long as the button is held down. In addition we could apply the same idea for moving the character to the target. So, instead of the character moving until it reaches the target after a button press, the character would only move while the attack button is held down.

**Comment by AJenbo** [comment]
It's not emulating a mouse click, if it was it would already be continuing to attack :) But yes this should be implemented for Gamepads as well.

I'm not sure how I feel about the second part, it's doable but I'm not sure it's strictly better.

**Reply by fwannmacher** [reply]
so i'll work on that and lets see how it will look like :)

**Comment by fwannmacher** [comment]
https://github.com/diasurgical/devilutionX/pull/4019
